File: Chapter_01_Introduction/Getting_Started_The_Easy_Problems/Super_Easy/10550-Combination_Lock/Main.java
/*
 * Problem 10550 - Combination Lock
 * Java version 6/11/2016
 */

import java.io.PrintStream;
import java.util.Scanner;

public class Main {
    static final Scanner input = new Scanner(System.in);
    static final PrintStream output = System.out;

    public static void main(String[] args) {

        StringBuffer results = new StringBuffer();
        boolean first = true;

        while (true) {
            int start = input.nextInt();
            int a = input.nextInt();
            int b = input.nextInt();
            int c = input.nextInt();

            if (start == 0 && a == 0 && b == 0 && c == 0) {
                break;
            }

            // clockwise from the start
            int ticks = start - a;
            if (ticks < 0) {
                ticks += 40;
            }
            int totalTicks = ticks;
    
            // counter clockwise
            ticks = b - a;
            if (ticks < 0) {
                ticks += 40;
            }
            totalTicks += ticks;

            // clockwise
            ticks = b - c;
            if (ticks < 0) {
                ticks += 40;
            }
            totalTicks += ticks;

            results.append(1080 + (totalTicks * 9) + "\n");
        }
        System.out.print(results.toString());
    }
}

File: Chapter_01_Introduction/Getting_Started_The_Easy_Problems/Super_Easy/10550-Combination_Lock/main.cpp
/*
 * Problem 10550 - Combination Lock
 */
#include <iostream>
#include <sstream>

int countTicks(int a, int b) {
    int ticks {a - b};
    return (ticks < 0 ? ticks + 40 : ticks);
}

int main() {
    std::ios::sync_with_stdio(0);
    std::cin.tie(0);

    constexpr int DEGREES_PER_TICK {9};
    constexpr int EXTRA_TURNS {1080};

    std::ostringstream output;
    while (true) {
        int start, a, b, c;
        std::cin >> start >> a >> b >> c;
        if (start == 0 && a == 0 && b == 0 && c == 0) {
            break;
        }

        // first turn counterclockwis
        int totalTicks {countTicks(start, a)};

        // second turn clockwise
        totalTicks += countTicks(b, a);

        // thrid turn counterclockwise
        totalTicks += countTicks(b, c);

        // convert ticks to degrees and add in the constant extra turns
        output << (EXTRA_TURNS + (DEGREES_PER_TICK * totalTicks)) << '\n';
    }
    std::cout << output.str();
}
